test/hotspot/jtreg/serviceability/jvmti/events/NotifyFramePopStressTest/NotifyFramePopStressTest.java

```diff
@@ -0,0 +1,136 @@
+        try {
+            System.loadLibrary("NotifyFramePopStressTest");
+        } catch (UnsatisfiedLinkError ex) {
+            System.err.println("Could not load NotifyFramePopStressTest library");
+            System.err.println("java.library.path:"
+                + System.getProperty("java.library.path"));
+            throw ex;
+        }
+    }
+
+    public static void main(String args[]) {
+        if (!canGenerateFramePopEvents()) {
+            throw new SkippedException("FramePop event is not supported");
+        }
+        Thread testThread = Thread.currentThread();
+        Thread controlThread = new Thread(() -> control(testThread), "Control Thread");
+
+        setFramePopNotificationMode(testThread, true);
+        controlThread.start();
+        sleep(10);
+
+        for (int i = 0; i < 10*1000; i++) {
+            foo();
+            bar();
+        }
+        done = true;
+
+        try {
+            controlThread.join();
+        } catch (InterruptedException e) {
+        }
+
+        if (failed()) {
+            throw new RuntimeException("Test FAILED: see log for details");
+        } else {
+            log("Test PASSED");
+        }
+    }
+
+    private static void sleep(long ms) {
+        try {
+            Thread.sleep(ms);
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private static void control(Thread thread) {
+        System.out.println("control has started");
+        while (!done) {
+            suspend(thread);
+            if (notifyFramePop(thread)) {
+                notifyCount++;
+                System.out.println("control incremented notifyCount to " + notifyCount);
+            }
+            resume(thread);
+            int waitCount = 0;
+            while (notifyCount != getPopCount()) {
+                sleep(1);
+                waitCount++;
+                if (waitCount > 1000) {
+                    break;
+                }
+            }
+            if (waitCount > 50) {
+                System.out.println("About to fail. notifyCount=" + notifyCount +
+                                   " getPopCount()=" + getPopCount());
+                throw new RuntimeException("Test FAILED: Waited too long for notify: " + waitCount);
+            }
+        }
+        System.out.println("control has finished: " + notifyCount);
+    }
+
+    private native static void suspend(Thread thread);
+    private native static void resume(Thread thread);
+    private native static int getPopCount();
+    private native static boolean failed();
+    private native static boolean canGenerateFramePopEvents();
+    private native static void setFramePopNotificationMode(Thread thread, boolean enabled);
+    private native static boolean notifyFramePop(Thread thread);
+
+    private static void log(String msg) {
+        System.out.println(msg);
+    }
+
+    private static int fetchInt() {
+        return 13;
+    }
+
+    private static int foo() {
+        return fetchInt();
```

Comment:
The test checks name of the method being popped.
I think it would be better to call different methods from `foo()` and `bar()`

Reply:
Fixed, thanks.